### PR TITLE
Add support for case insensitive enumerations

### LINKED ISSUE #2559: Support CamelCase CSS names
Following on from a discussion on the mailling list, https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/BVAPlqMyI1Y.

@bryevdv mentioned that "One thing to mention is that "Black" (capital B) is not recognized as a color, all color names are lower case."

But:
* The link we point to in the docs: http://www.w3schools.com/cssref/css_colornames.asp
* The link we go to in our OpenURL example: http://www.colors.commutercreative.com/
* And, the OpenURL example: https://github.com/bokeh/bokeh/blob/master/examples/glyphs/colors.py
 
all use CamelCase colors. 

I have been trying to use them and just assumed it was broken. 

Turns out we need to lowercase them - I can't see this in the docs.

Either we need a warning for this, or, and I think better, we should support the CamelCase which seems to be the convention.

### PR DESCRIPTION
**Pull request by mattpap**
Effectively this allows now colors to match CSS specification.

fixes #2559

### CODE REVIEW

File: bokeh/enums.py [hunk 1]
```diff
@@ -7,9 +7,20 @@
 from . import colors, icons, palettes
 
 class Enumeration(object):
-    pass
 
-def enumeration(*values):
+    __slots__ = []
```
Comment (bryevdv) [inline bokeh/enums.py]: I always prefer tuples for slots.

**Comment by bryevdv** [comment]
Minor comment otherwise LGTM

**Comment by birdsarah** [comment]
Hi, I'm running the sample example from #2559. And it's not working for me.

````python
from bokeh.io import output_notebook, show
from bokeh.plotting import figure

output_notebook()
f = figure(x_range=(0,3), y_range=(0,4))
f.circle([1], [1], radius=0.2, fill_color=["LemonChiffon"])  # Works
f.circle([2], [1], radius=0.2, fill_color=["lemonchiffon"])  # Works

f.circle([1], [2], radius=0.2, fill_color="LemonChiffon")  # Does not work
f.circle([2], [2], radius=0.2, fill_color="lemonchiffon")  # Works
show(f)
````

I'm assuming its working for you, what information can I give to help track this down further.

**Comment by damianavila** [comment]
I am seeing the same as @birdsarah, ping @mattpap

File: bokeh/enums.py (hunk 1 above)
Comment (mattpap) [inline bokeh/enums.py]: Why?

**Comment by mattpap** [comment]
> And it's not working for me.

That's the effect of lack of proper abstractions in the original code and me not testing your example.

**Comment by mattpap** [comment]
@damianavila, should work now.

**Comment by fpliger** [comment]
@mattpap it's not working for me... That example raises the following on the browser:

```Uncaught Error: color property 'fill_color' given invalid value: LemonChiffon```

File: bokeh/enums.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/enums.py]: Just preference. Since changing `__slots__` (if you do it) has no effect, I prefer a type that can't be changed.

**Comment by damianavila** [comment]
>That example raises the following on the browser:

Ditto here...

**Comment by mattpap** [comment]
@damianavila, how about now? Forgot to update bokehjs as well.

**Comment by damianavila** [comment]
Works now... thanks @mattpap 
LGTM, merging now...

**Pull request merged**